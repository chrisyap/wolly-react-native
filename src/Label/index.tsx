import React from 'react';
import Txt, { TxtProps } from '../Text';
import { isTablet } from '../Util';

export interface LabelProps extends TxtProps {
  required?: boolean;
  uppercase?: boolean;
}

const Label: React.FC<LabelProps> = (props) => (
  <Txt fontSize={isTablet ? 7 : props.uppercase ? 5 : 6} uppercase={props.uppercase} color="secondary" {...props}>
    {props.children} {props.required && '*'}
  </Txt>
);

export default Label;
