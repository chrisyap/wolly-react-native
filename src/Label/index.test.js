import React from 'react';
import Label from './index';

describe('Label', () => {
  const defaultProps = {
    children: 'Text goes here',
  };
  it('renders correctly', () => {
    const wrapper = shallow(<Label {...defaultProps} />)
      .dive()
      .dive();
    expect(wrapper).toMatchSnapshot();
  });
  it('renders with required prop correctly', () => {
    const props = {
      ...defaultProps,
      required: true,
    };
    const wrapper = shallow(<Label {...props} />)
      .dive()
      .dive();
    expect(wrapper).toMatchSnapshot();
  });
  it('renders with uppercase text correctly', () => {
    const props = {
      ...defaultProps,
      uppercase: true,
    };
    const wrapper = shallow(<Label {...props} />)
      .dive()
      .dive();
    expect(wrapper).toMatchSnapshot();
  });
});
