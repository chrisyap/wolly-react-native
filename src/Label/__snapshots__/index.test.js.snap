// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Label renders with required prop correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Label renders with uppercase text correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
